CMakeLists.txt

```diff
@@ -243,47 +243,47 @@ COLMAP_ADD_SOURCE_DIR(src/thirdparty/PoissonRecon THIRDPARTY_POISSON_RECON_SRCS
 COLMAP_ADD_SOURCE_DIR(src/thirdparty/SiftGPU THIRDPARTY_SIFT_GPU_SRCS *.h *.cpp *.cu)
 COLMAP_ADD_SOURCE_DIR(src/thirdparty/VLFeat THIRDPARTY_VLFEAT_SRCS *.h *.c *.tc)
 
-# Add all of the source files to a regular library target, as using a custom
-# target does not allow us to set its C++ include directories (and thus
-# intellisense can't find any of the included files).
-set(ALL_SRCS
-    ${CONTROLLERS_SRCS}
-    ${ESTIMATORS_SRCS}
-    ${EXE_SRCS}
-    ${FEATURE_SRCS}
-    ${GEOMETRY_SRCS}
-    ${IMAGE_SRCS}
-    ${MATH_SRCS}
-    ${MVS_SRCS}
-    ${OPTIM_SRCS}
-    ${RETRIEVAL_SRCS}
-    ${SCENE_SRCS}
-    ${SENSOR_SRCS}
-    ${SFM_SRCS}
-    ${TOOLS_SRCS}
-    ${UI_SRCS}
-    ${UTIL_SRCS}
-    ${THIRDPARTY_POISSON_RECON_SRCS}
-    ${THIRDPARTY_SIFT_GPU_SRCS}
-    ${THIRDPARTY_VLFEAT_SRCS}
-)
-
-if(LSD_ENABLED)
-    list(APPEND ALL_SRCS
-        ${THIRDPARTY_LSD_SRCS}
-    )
-endif()
-
-add_library(
-    ${COLMAP_SRC_ROOT_FOLDER}
-    ${ALL_SRCS}
-)
-
-# Prevent the library from being compiled automatically.
-set_target_properties(
-    ${COLMAP_SRC_ROOT_FOLDER} PROPERTIES
-    EXCLUDE_FROM_ALL 1
-    EXCLUDE_FROM_DEFAULT_BUILD 1)
+# # Add all of the source files to a regular library target, as using a custom
```

Comment:
I think we want to retain the option to generate this target for Visual Studio / XCode users. How about introducing a new option that is off by default?